test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -282,13 +282,94 @@ object Assertion extends AssertionVariants {
     }
 
   /**
-   * Makes a new assertion that requires an iterable contain the specified
-   * element. See [[Assertion.exists]] if you want to require an iterable to contain an element
+   * Makes a new assertion that requires a numeric value is positive.
+   */
+  def isPositive[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("isPositive")()(actual => num.gt(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is non negative.
+   */
+  def nonNegative[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("nonNegative")()(actual => num.gteq(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is negative.
+   */
+  def isNegative[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("isNegative")()(actual => num.lt(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is non positive.
+   */
+  def nonPositive[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("nonPositive")()(actual => num.lteq(actual, num.zero))
+
+  /**
+   * Makes a new assertion that requires a numeric value is zero.
+   */
+  def isZero[A](implicit num: Numeric[A]): Assertion[A] =
+    Assertion.assertion("isZero")()(_ == num.zero)
+
+  /**
+   * Makes a new assertion that requires a value to be equal to one of the specified values.
+   */
+  def equalToOneOf[A, B](value: A, other: A, remaining: A*)(implicit eql: Eql[A, B]): Assertion[B] = {
```

Comment:
Maybe `isOneOf`? I would make this take a collection instead of using variable arguments.

Reply:
Good idea. For the collection type, I think `Set` would be a good idea to avoid passing in duplicate elements, i.e. `isOneOf(List(1, 1, 2))`.